packages/firestore/test/unit/local/indexeddb_persistence.test.ts

```diff
@@ -712,6 +712,109 @@ describe('IndexedDbSchema: createOrUpgradeDb', () => {
     });
   });
 
+  it('can use read-time index after schema migration', async () => {
+    // This test creates a database with schema version 8 that has a few
+    // remote documents, adds an index and then reads new documents back
+    // via that index.
+
+    const existingDocPaths = [
+      'coll1/doc1',
+      'coll1/doc2',
+      'coll2/doc1',
+      'coll2/doc2'
+    ];
+    const newDocPaths = [
+      'coll1/doc3',
+      'coll1/doc4',
+      'coll2/doc3',
+      'coll2/doc4'
+    ];
+
+    await withDb(8, db => {
+      const sdb = new SimpleDb(db);
+      return sdb.runTransaction('readwrite', V8_STORES, txn => {
+        const remoteDocumentStore = txn.store<
+          DbRemoteDocumentKey,
+          DbRemoteDocument
+        >(DbRemoteDocument.store);
+
+        // Write the remote document entries.
+        return PersistencePromise.forEach(existingDocPaths, (path: string) => {
+          const remoteDoc = doc(path, /*version=*/ 1, { data: 1 });
+
+          const dbRemoteDoc = TEST_SERIALIZER.toDbRemoteDocument(
+            remoteDoc,
+            remoteDoc.version
+          );
+          // Mimic the old serializer and delete previously unset values
+          delete dbRemoteDoc.readTime;
+          delete dbRemoteDoc.parentPath;
+
+          return remoteDocumentStore.put(
+            remoteDoc.key.path.toArray(),
+            dbRemoteDoc
+          );
+        });
+      });
+    });
+
+    // Migrate to v8 and verify that new documents are indexed.
+    await withDb(9, db => {
+      const sdb = new SimpleDb(db);
+      return sdb.runTransaction('readwrite', V8_STORES, txn => {
+        const remoteDocumentStore = txn.store<
+          DbRemoteDocumentKey,
+          DbRemoteDocument
+        >(DbRemoteDocument.store);
+
+        // Verify the existing remote document entries.
+        return PersistencePromise.forEach(existingDocPaths, (path: string) => {
+          const remoteDoc = doc(path, /*version=*/ 1, { data: 1 });
```

Comment:
It seems we construct this and then just use remoteDoc.key.path?  Why not just construct a ResourcePath directly?

Reply:
It adds the elusion of complexity. I thought you might like it. I removed it, but we can always add it back :)